diff --git a/test/rust/unittest/rust_code_emitter_test/BUILD.gn b/test/rust/unittest/rust_code_emitter_test/BUILD.gn
index 45ba14b..01a8ffe 100644
--- a/test/rust/unittest/rust_code_emitter_test/BUILD.gn
+++ b/test/rust/unittest/rust_code_emitter_test/BUILD.gn
@@ -19,7 +19,6 @@ config("idl_unittest_test_config") {
   include_dirs = [
     "../../common/",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
@@ -83,7 +82,7 @@ ohos_unittest("code_emitter_test") {
   configs = [ ":idl_unittest_test_config" ]
   sources = [ "code_emitter_test.cpp" ]
   sources += common_sources
-  deps = [ "//third_party/bounds_checking_function:libsec_static" ]
+  external_deps = [ "bounds_checking_function:libsec_shared" ]
 }
 
 group("unittest") {
diff --git a/test/ts/moduletest/ts_code_emitter_test/BUILD.gn b/test/ts/moduletest/ts_code_emitter_test/BUILD.gn
index 8e3e799..e6f332f 100644
--- a/test/ts/moduletest/ts_code_emitter_test/BUILD.gn
+++ b/test/ts/moduletest/ts_code_emitter_test/BUILD.gn
@@ -19,7 +19,6 @@ config("idl_module_test_config") {
   include_dirs = [
     "../../common",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
@@ -82,7 +81,7 @@ ohos_moduletest("ts_code_emitter_test") {
   configs = [ ":idl_module_test_config" ]
   sources = [ "ts_code_emitter_test.cpp" ]
   sources += common_sources
-  deps = [ "//third_party/bounds_checking_function:libsec_static" ]
+  external_deps = [ "bounds_checking_function:libsec_shared" ]
 }
 
 group("moduletest") {
diff --git a/test/ts/unittest/ts_code_emitter_interface_test/BUILD.gn b/test/ts/unittest/ts_code_emitter_interface_test/BUILD.gn
index 70fb88a..dbf8842 100644
--- a/test/ts/unittest/ts_code_emitter_interface_test/BUILD.gn
+++ b/test/ts/unittest/ts_code_emitter_interface_test/BUILD.gn
@@ -19,7 +19,6 @@ config("idl_unittest_test_config") {
   include_dirs = [
     "../../common/",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
@@ -82,7 +81,7 @@ ohos_unittest("ts_code_emitter_interface_test") {
   configs = [ ":idl_unittest_test_config" ]
   sources = [ "ts_code_emitter_interface_test.cpp" ]
   sources += common_sources
-  deps = [ "//third_party/bounds_checking_function:libsec_static" ]
+  external_deps = [ "bounds_checking_function:libsec_shared" ]
 }
 
 group("unittest") {
diff --git a/test/ts/unittest/ts_code_emitter_proxy_test/BUILD.gn b/test/ts/unittest/ts_code_emitter_proxy_test/BUILD.gn
index 8604009..6a65d2b 100644
--- a/test/ts/unittest/ts_code_emitter_proxy_test/BUILD.gn
+++ b/test/ts/unittest/ts_code_emitter_proxy_test/BUILD.gn
@@ -20,7 +20,6 @@ config("idl_unittest_test_config") {
     "../../common",
     "${IDL_DIR}/",
     "${IDL_DIR}/codegen/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
@@ -86,11 +85,7 @@ ohos_unittest("ts_code_emitter_proxy_test") {
   sources = [ "ts_code_emitter_proxy_test.cpp" ]
   sources += idl_common_sources
 
-  deps = [
-    "//third_party/bounds_checking_function:libsec_static",
-    "//third_party/googletest:gmock_main",
-    "//third_party/googletest:gtest_main",
-  ]
+  external_deps = [ "bounds_checking_function:libsec_shared" ]
 }
 
 group("unittest") {
diff --git a/test/ts/unittest/ts_code_emitter_stub_test/BUILD.gn b/test/ts/unittest/ts_code_emitter_stub_test/BUILD.gn
index da81ae2..85f26ea 100644
--- a/test/ts/unittest/ts_code_emitter_stub_test/BUILD.gn
+++ b/test/ts/unittest/ts_code_emitter_stub_test/BUILD.gn
@@ -19,7 +19,6 @@ config("idl_unittest_test_config") {
   include_dirs = [
     "../../common",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
@@ -82,7 +81,7 @@ ohos_unittest("ts_code_emitter_stub_test") {
   configs = [ ":idl_unittest_test_config" ]
   sources = [ "ts_code_emitter_stub_test.cpp" ]
   sources += common_sources
-  deps = [ "//third_party/bounds_checking_function:libsec_static" ]
+  external_deps = [ "bounds_checking_function:libsec_shared" ]
 }
 
 group("unittest") {
diff --git a/test/unittest/metadata_dumper_test/BUILD.gn b/test/unittest/metadata_dumper_test/BUILD.gn
index 8d96410..49ab421 100644
--- a/test/unittest/metadata_dumper_test/BUILD.gn
+++ b/test/unittest/metadata_dumper_test/BUILD.gn
@@ -23,7 +23,6 @@ config("idl_unittest_test_config") {
   include_dirs = [
     "../../common/",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
diff --git a/test/unittest/metadata_reader_test/BUILD.gn b/test/unittest/metadata_reader_test/BUILD.gn
index 0e4aed0..b2b4dcd 100644
--- a/test/unittest/metadata_reader_test/BUILD.gn
+++ b/test/unittest/metadata_reader_test/BUILD.gn
@@ -23,7 +23,6 @@ config("idl_unittest_test_config") {
   include_dirs = [
     "../../common/",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
diff --git a/test/unittest/metadata_serializer_test/BUILD.gn b/test/unittest/metadata_serializer_test/BUILD.gn
index 301e4a8..6369af0 100644
--- a/test/unittest/metadata_serializer_test/BUILD.gn
+++ b/test/unittest/metadata_serializer_test/BUILD.gn
@@ -23,7 +23,6 @@ config("idl_unittest_test_config") {
   include_dirs = [
     "../../common/",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
diff --git a/test/unittest/util_options_test/BUILD.gn b/test/unittest/util_options_test/BUILD.gn
index 8fd681e..662dafc 100644
--- a/test/unittest/util_options_test/BUILD.gn
+++ b/test/unittest/util_options_test/BUILD.gn
@@ -23,7 +23,6 @@ config("idl_unittest_test_config") {
   include_dirs = [
     "../../common/",
     "${IDL_DIR}/",
-    "//third_party/bounds_checking_function/include/",
   ]
 }
 
